File: java/sfo_21_adjust_the_order_of_numbers_in_an_array_s1/sfo_21_adjust_the_order_of_numbers_in_an_array_s1.java
/*
* File: sfo_21_adjust_the_order_of_numbers_in_an_array_s1.java
* Created Time: 2021-12-09
* Author: Krahets (krahets@163.com)
*/

package sfo_21_adjust_the_order_of_numbers_in_an_array_s1;

import include.*;
import java.util.*;

// ===== Solution Code =====
class Solution {
    public int[] exchange(int[] nums) {
        int i = 0, j = nums.length - 1, tmp;
        while(i < j) {
            while(i < j && (nums[i] & 1) == 1) i++;
            while(i < j && (nums[j] & 1) == 0) j--;
            tmp = nums[i];
            nums[i] = nums[j];
            nums[j] = tmp;
        }
        return nums;
    }
}

public class sfo_21_adjust_the_order_of_numbers_in_an_array_s1 {
    public static void main(String[] args) {
        // ======= Test Case =======
        int[] nums = { 1, 2, 3, 4 };
        // ====== Driver Code ======
        Solution slt = new Solution();
        int[] res = slt.exchange(nums);
        System.out.println(Arrays.toString(res));
    }
}

File: python/sfo_21_adjust_the_order_of_numbers_in_an_array_s1.py
'''
File: sfo_21_adjust_the_order_of_numbers_in_an_array_s1.py
Created Time: 2021-12-09
Author: Krahets (krahets@163.com)
'''

from include import *

# ===== Solution Code =====
class Solution:
    def exchange(self, nums: List[int]) -> List[int]:
        i, j = 0, len(nums) - 1
        while i < j:
            while i < j and nums[i] & 1 == 1: i += 1
            while i < j and nums[j] & 1 == 0: j -= 1
            nums[i], nums[j] = nums[j], nums[i]
        return nums

# ======= Test Case =======
nums = [1, 2, 3, 4]
# ====== Driver Code ======
slt = Solution()
res = slt.exchange(nums)
print(res)
